### PR TITLE
MH-13685: Added dublincore, security to publish (merge) and replublish-oaimph

### PR DESCRIPTION
**Pull request by TurRil**
Added dublincore and security to publish (merge) and replublish-oaimph, and small ui updates to use mediapackage.start or dcCreated.

This fixes the bug where dcCreated is not filled in correctly and the ACL's are not merged in for events where a caption gets added from an attachment workflow.

### CODE REVIEW

**Comment by gregorydlogan** [comment]
As annoying as it would be, this should be two issues.  That being said, this is filed.  Next time.

Regarding the ACL changes, these seem fine to me.  I don't see anything obvious that prevents a merge.
Regarding the dcCreated field, this is more problematic.  We discussed a bunch of dc and (specifically) datetime related things this morning.  @wsmirnow had things to say about this.  Would this cause issues with, or fix things you're looking at.  cc @karendolan

File: etc/workflows/attach-watson-transcripts.xml [hunk 1]
```diff
@@ -33,12 +33,24 @@
       exception-handler-workflow="partial-error"
       description="Distribute and publish to engage server">
       <configurations>
+        <configuration key="download-source-flavors">dublincore/*,security/*</configuration>
```
Comment (wsmirnow) [inline etc/workflows/attach-watson-transcripts.xml:36]: Why are you updating dublincore and security catalogs here? The `attach-watson-transcription` operation should not touch any metadata.

File: etc/workflows/attach-watson-transcripts.xml [hunk 2]
```diff
@@ -33,12 +33,24 @@
...
         <configuration key="check-availability">true</configuration>
       </configurations>
     </operation>
 
+    <operation
+      id="republish-oaipmh"
+      exception-handler-workflow="partial-error"
+      description="Update recording metadata in default OAI-PMH repository">
+      <configurations>
+        <configuration key="source-flavors">dublincore/*,security/*</configuration>
```
Comment (wsmirnow) [inline etc/workflows/attach-watson-transcripts.xml:48]: Same here… Removing this line should leave the metadata in the oai-pmh repository untouched but add/update all `download-source-tags` marked assets.

File: etc/workflows/publish-uploaded-assets.xml [hunk 3]
```diff
@@ -49,7 +49,8 @@
       exception-handler-workflow="partial-error"
       description="Update recording in Opencast Media Module">
       <configurations>
-        <configuration key="download-source-flavors">${download-source-flavors}</configuration>
+        <configuration key="download-source-flavors">${download-source-flavors},dublincore/*,security/*</configuration>
```
Comment (wsmirnow) [inline etc/workflows/publish-uploaded-assets.xml:52]: Updating already published metadata on asset upload isn't intuitive. Here you only want to add new asset to the publication.

File: etc/workflows/publish-uploaded-assets.xml [hunk 4]
```diff
@@ -63,7 +64,8 @@
       exception-handler-workflow="partial-error"
       description="Update recording metadata in default OAI-PMH repository">
       <configurations>
-        <configuration key="source-flavors">${download-source-flavors}</configuration>
+       <configuration key="source-flavors">${download-source-flavors},dublincore/*,security/*</configuration>
```
Comment (wsmirnow) [inline etc/workflows/publish-uploaded-assets.xml:67]: See above…

**Review by wsmirnow — CHANGES_REQUESTED** [review]

**Comment by wsmirnow** [comment]
> As annoying as it would be, this should be two issues. That being said, this is filed. Next time.
> 
> Regarding the ACL changes, these seem fine to me. I don't see anything obvious that prevents a merge.
> Regarding the dcCreated field, this is more problematic. We discussed a bunch of dc and (specifically) datetime related things this morning. @wsmirnow had things to say about this. Would this cause issues with, or fix things you're looking at. cc @karendolan

I've only touched the Admin UI. My PR #1079 does not conflict with this PR.

**Comment by karendolan** [comment]
I notice that dcTitle is also not guaranteed to exist and has a fallback value, per the JavaScript. I wonder why the assumption that mediapackage "start" will exist if the dcCreated does not. How are @TurRil 's mediapackages getting compiled and published without some core metadata?    
> Regarding the dcCreated field, this is more problematic. We discussed a bunch of dc and (specifically) datetime related things this morning.

**Comment by wsmirnow** [comment]
@TurRil, I would splitt this PR into two seperate PRs. The second part (theodul model start date fallback) is fine for me. But…

Why do you want to update dc and security catalogs on transcriptions add/update? Does the transcription operation touch any data of the dublincore catalog? The ACLs shouldn't be touched anyway.

**Comment by KatrinIhler** [comment]
@TurRil Conflicts!

**Comment by karendolan** [comment]
@TurRil I don't understand why this pull adds the dublincore and ACLS to be publish during an upload asset and attach caption workflow. Are you seeing those files being removed during your attach caption or upload asset workflows? Does your site have a process that modifies those files when captions or other attachments are added to a mediapackage?

Our site uses the upload asset and attach caption workflows every day in our modified Opencast v5/6/7/8x and have had no problems with existing published dublincore or ACLs.

**Comment by gregorydlogan** [comment]
@TurRil any progress here?

**Comment by KatrinIhler** [comment]
@TurRil What's the status here? If there can't be any work done on this in the foreseeable future, it might be better to close this PR and reopen at a later date.

**Comment by KatrinIhler** [comment]
As per our recent discussion in the dev meeting and the proposal that followed, I'm going to close this PR since this has seen no updates within the past 1,5 years. This does not constitute a rejection, feel free to re-open once you have time to address the review findings. :)

**Pull request closed without merging**